ghw/libghw.c

```diff
@@ -367,11 +420,23 @@ ghw_read_str (struct ghw_handler *h)
 
   if (hdr[0] != 0 || hdr[1] != 0 || hdr[2] != 0 || hdr[3] != 0)
     return -1;
-  h->nbr_str = ghw_get_i32 (h, &hdr[4]);
+
+  // Read number of strings and total string size. Make sure that nbr_string
+  // is not too large, as it's incremented afterwards.
+  h->nbr_str = (uint32_t) ghw_get_i32 (h, &hdr[4]);
```

Comment:
Simply test that the value of returned by `ghw_get_i32` is not negative.

Reply:
Sure, but don't blame me if someone wants to trace 2^31 signals :wink: